torch/csrc/autograd/profiler.cpp

```diff
@@ -95,14 +97,25 @@ RecordFunction::RecordFunction(const char* name) {
   pushRange(name);
 }
 
+RecordFunction::RecordFunction(const char* name, int64_t current_sequence_nr) 
+{
+  if (state == ProfilerState::Disabled)
+    return;
+  std::stringstream s;
+  s << name << ", current sequence nr " << current_sequence_nr;
+  pushRange(s.str());
```

Comment:
All string formatting should happen when the trace is processed. Right now it happens in the hot path and greatly increases the overhead of the profiler.

Reply:
Again, thank you for the quick review.

Less verbose is definitely an improvement.  How about `Add, seq=123` and `AddBackward, forward_seq=123`?

`All string formatting should happen when the trace is processed.` what do you mean by this?  `trace` is an overloaded/ambiguous term here.  Should I defer the string processing to [the actual nvtxRangePushA call`](https://github.com/pytorch/pytorch/blob/ca57b19fd7ee94f62e9c9aae97a7687f257caa29/torch/csrc/autograd/profiler.cpp#L48), which is the latest it can be deferred?

The id needs to be dynamically appended to a string somehow, some way.  Globally, I don't see how this overhead can be avoided.